src/docfx/lib/json/JsonContractResolver.cs

```diff
@@ -30,13 +30,21 @@ protected override JsonObjectContract CreateObjectContract(Type objectType)
         protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
         {
             var prop = base.CreateProperty(member, memberSerialization);
-            ShouldNotSerializeEmptyArray();
+            ShouldNotSerialize();
```

Comment:
Consider having two methods: `DoNotSerializeEmptyArray` and `DoNotSerializeNullSourceInfo`